data/types/base/base.go

```diff
@@ -63,41 +68,29 @@ func (b *Base) Meta() interface{} {
 }
 
 func (b *Base) Parse(parser data.ObjectParser) error {
-	// b.Type = parser.ParseString("type")	// TODO_DATA: We do not parse out the type (this is set in the New function)
-	// b.UserID = parser.ParseString("userId")	// TODO_DATA: We do not parse UserID, we set this when we receive the data for the target user
+	b.Annotations = parser.ParseInterfaceArray("annotations")
+	b.ClockDriftOffset = parser.ParseInteger("clockDriftOffset")
+	b.ConversionOffset = parser.ParseInteger("conversionOffset")
 	b.DeviceID = parser.ParseString("deviceId")
-	b.Time = parser.ParseString("time")
-	// b.UploadID = parser.ParseString("uploadId") // TODO_DATA: We do not parse UploadID, we create and set this
 	b.DeviceTime = parser.ParseString("deviceTime")
-	b.TimezoneOffset = parser.ParseInteger("timezoneOffset")
-	b.ConversionOffset = parser.ParseInteger("conversionOffset")
-	b.ClockDriftOffset = parser.ParseInteger("clockDriftOffset")
 	b.Payload = parser.ParseInterface("payload")
-	b.Annotations = parser.ParseInterfaceArray("annotations")
-
-	// b.GroupID = parser.ParseString("_groupId")	// TODO_DATA: We do not parse GroupID, we set this when we receive the data for the target user
+	b.Time = parser.ParseString("time")
+	b.TimezoneOffset = parser.ParseInteger("timezoneOffset")
 
 	return nil
 }
 
 func (b *Base) Validate(validator data.Validator) error {
-	// validator.ValidateString("createdTime", &b.CreatedTime).LengthGreaterThanOrEqualTo(1)
-	// validator.ValidateString("type", &b.Type).LengthGreaterThanOrEqualTo(1)
-
-	// validator.ValidateString("userId", b.UserID).Exists().LengthGreaterThanOrEqualTo(10)	// TODO_DATA: Validation is for parsed data only
-	validator.ValidateString("deviceId", b.DeviceID).Exists().LengthGreaterThanOrEqualTo(1)
+	// validator.ValidateInterfaceArray("annotations", b.Annotations)    // TODO: Any validations? Optional? Size?
```

Comment:
Yes, there are a number of TODOs throughout the project. We'll need to at least revisit each one and decide whether it is something that needs to be addressed before prd or not.